### DISCUSSION TITLE
JSON output file is not valid JSON

### DISCUSSION THREAD

**Discussion by 0xcrypto** [projectdiscovery/nuclei#849] [General]
**Describe the bug**
```
nuclei -target hackberry.xyz -json -o /tmp/result.json
```

The resulting JSON is not valid as the objects are not packed within an array.

**Nuclei version**: 2.4.0

**Comment by CharlieEriksen** [comment]
It's valid [jsonlines](https://jsonlines.org/).

If you change the json file output, you'll also now have two different output formats. The stdout output is jsonlines, and that by its nature will not be an array. Diverging in behaviour between stdout and output file format would be confusing.

**Comment by 0xcrypto** [comment]
> It's valid [jsonlines](https://jsonlines.org/).
> 
> If you change the json file output, you'll also now have two different output formats. The stdout output is jsonlines, and that by its nature will not be an array. Diverging in behaviour between stdout and output file format would be confusing.

From the help command, 
```
   -json                               Write json output to files
```

So how do I get json in array and not jsonlines?

**Comment by CharlieEriksen** [comment]
My recommendation would be: 
`jq --slurp . < nuclei-output.json > nuclei-output-array.json`

In which case, the issue here is that the help text needs to specify that it's jsonl, not json.

**Reply by 0xcrypto** [reply]
Thanks, I will try that. Yeah help text should be a little more specific.

**Comment by ehsandeep** [comment]
Moving to the discussion board as it may help others.
